packages/manhattan/src/index.js

```diff
@@ -1,42 +1,46 @@
-/* eslint-disable react/prop-types */
 import React from 'react'
 import PropTypes from 'prop-types'
-import R from 'ramda'
 import { scaleLinear } from 'd3-scale'
-import { max, min } from 'd3-array'
+import { extent } from 'd3-array'
 
 import {
   HUMAN_CHROMOSOMES,
   HUMAN_AUTOSOMES,
 } from '@broad/utilities/src/constants'
 
+
+function randomColor() {
+  const r = Math.floor(Math.random() * 256)
+  const g = Math.floor(Math.random() * 256)
+  const b = Math.floor(Math.random() * 256)
+  return `rgb(${r},${g},${b})`
+}
+
+
 const ManhattanPlot = ({
   data,
-  title = '',
-  width = 900,
-  height = 500,
-  sexChromosomes = false,
-  showAxisBounds = false,
+  title,
+  width,
+  height,
+  includeSexChromosomes,
+  showAxisBounds,
 }) => {
-  console.log(data)
   const padding = 60
-  const yData = R.pluck('-log10p', data)
 
-  const plotChromosomes = sexChromosomes ? HUMAN_CHROMOSOMES : HUMAN_AUTOSOMES
-  const rgb = () => Math.floor(Math.random() * 256)
-  const colorCode = () => `rgb(${rgb()}, ${rgb()}, ${rgb()})`
+  const plotChromosomes = includeSexChromosomes ? HUMAN_CHROMOSOMES : HUMAN_AUTOSOMES
   const chromosomeColors = plotChromosomes.reduce((acc, chr) =>
-    ({ ...acc, [chr]: colorCode() }), {})
+    ({ ...acc, [chr.replace('chr', '')]: randomColor() }), {})
 
   const xScale = scaleLinear()
     .domain([0, data.length])
     .range([0 + padding, width])
 
+  const yExtent = extent(data, d => d['-log10p'])
   const yScale = scaleLinear()
-    .domain([min(yData), max(yData) + (max(yData) * 0.1)])
+    .domain([yExtent[0], yExtent[1] * 1.1])
```

Comment:
Previously, this iterated over `data` 4 times (once with `R.pluck`, once with `d3.min`, and twice with `d3.max`). Using `d3.extent` with an accessor reduces that to one iteration.

Reply:
Nice, didn't know about extent